src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -112,20 +112,44 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         if ((*so)[sfAmount] > (*bo)[sfAmount])
             return tecINSUFFICIENT_PAYMENT;
 
-        // If the buyer specified a destination, that destination must be
-        // the seller or the broker.
+        // If the buyer specified a destination
         if (auto const dest = bo->at(~sfDestination))
         {
-            if (*dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            // fixNFTokenBrokerAccept: Disabled
+            // the destination must be the seller or the broker.
+            if (!ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            {
                 return tecNFTOKEN_BUY_SELL_MISMATCH;
+            }
+
+            // fixNFTokenBrokerAccept: Enabled
+            // the destination must be the tx account
+            if (ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != ctx.tx[sfAccount])
+            {
+                return tecNO_PERMISSION;
+            }
         }
 
-        // If the seller specified a destination, that destination must be
-        // the buyer or the broker.
+        // If the seller specified a destination
         if (auto const dest = so->at(~sfDestination))
         {
-            if (*dest != bo->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            // fixNFTokenBrokerAccept: Disabled
+            // the destination must be the buyer or the broker.
+            if (!ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != bo->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            {
                 return tecNFTOKEN_BUY_SELL_MISMATCH;
+            }
+
+            // fixNFTokenBrokerAccept: Enabled
+            // the destination must be the tx account
+            if (ctx.view.rules().enabled(fixNFTokenBrokerAccept) &&
+                *dest != ctx.tx[sfAccount])
+            {
+                return tecNO_PERMISSION;
```

Comment:
Curious why would we use `tecNO_PERMISSION` instead of `tecNFTOKEN_BUY_SELL_MISMATCH` in this case